### ISSUE TITLE
"No webmention targets" after first reply in a thread

### ISSUE THREAD

**Issue by davidcelis** [snarfed/bridgy#1694]
I decided to open this separately from #1423 since there's probably a different issue happening here with threaded replies; in the linked issue, I was getting webmentions but without the target URL in the source HTML but, in this case, sometimes threaded replies stop targeting my original post and brid.gy doesn't know where to send them.

The most recent example happened with [this post](https://davidcel.is/notes/1769162483443147452) and a [thread of replies on Mastodon](https://xoxo.zone/@Mxzzyzx@gaygeek.social/112108429466543760). The linked reply [came through as a webmention just fine](https://brid.gy/log?module=background&start_time=1710639857&key=agdicmlkLWd5cjMLEghSZXNwb25zZSIldGFnOnhveG8uem9uZSwyMDEzOjExMjEwODQyOTQ2NjU0Mzc2MAw) but nothing else in that thread of replies was able to; I just see "No webmention targets" in brid.gy. Unfortunately, there don't appear to be logs either (I checked, but just got a "No log found!" page for the three that got listed:

* https://brid.gy/log?module=background&start_time=1710641889&key=agdicmlkLWd5cjMLEghSZXNwb25zZSIldGFnOnhveG8uem9uZSwyMDEzOjExMjEwODU4MjQ4OTA4OTk4OQw
* https://brid.gy/log?module=background&start_time=1710641889&key=agdicmlkLWd5cjMLEghSZXNwb25zZSIldGFnOnhveG8uem9uZSwyMDEzOjExMjEwODU3NzI2OTA3NTg1Ngw
* https://brid.gy/log?module=background&start_time=1710641889&key=agdicmlkLWd5cjMLEghSZXNwb25zZSIldGFnOnhveG8uem9uZSwyMDEzOjExMjEwODU3ODMzOTg2NTQ4NQw

I tried re-sending them, but brid.gy ended up trying to send them as webmentions to the _other_ person in the conversation ([example log](https://brid.gy/log?module=background&start_time=1710689198&key=agdicmlkLWd5cjMLEghSZXNwb25zZSIldGFnOnhveG8uem9uZSwyMDEzOjExMjEwODU4MjQ4OTA4OTk4OQw))

Based on what I see in brid.gy, it looks like my first reply in this thread wasn't caught by brid.gy and I'm wondering if that's at play here, causing everything else to be disconnected and not deliverable to me? https://xoxo.zone/@davidcelis/112108570378508887

**Comment by snarfed** [comment]
Hmm! Thanks for reporting, I'll look soon.

**Comment by snarfed** [comment]
I suspect this is a variation of https://github.com/snarfed/bridgy/issues/217#issuecomment-48375596 , which we've struggled with on and off for a long time. I think we fixed it pretty conclusively for Twitter, and some other silos, but maybe not Mastodon? Not sure. Ugh.

**Comment by snarfed** [comment]
Not 100% sure, but I'd guess that's ^ the root cause of https://github.com/snarfed/bridgy/issues/1423#issuecomment-1987724587 too.

**Comment by davidcelis** [comment]
FWIW, it looks like this problem isn't restricted to Mastodon; it also happens sometimes for my posts on Bluesky. Here's a recent example:

https://bsky.app/profile/davidcel.is/post/3kob3jwsooh24

Unfortunately, this is another case where I get a "No log found!" message so I'm not sure what other kind of info I can provide. From what I can tell, when this happens, it's one of my own replies that breaks the thread. E.g:

1. I post: https://bsky.app/profile/davidcel.is/post/3koayiw2m3c2t
2. Someone replies: https://bsky.app/profile/denverchick.bsky.social/post/3kob2eoc2z22g
3. I respond to the reply: https://bsky.app/profile/davidcel.is/post/3kob3jwsooh24

At that point (my own first nested reply), brid.gy is unable to continue the thread. Of course, this doesn't happen every time. I do notice it maybe once a week or so, though, and it's unfortunately not platform specific. Please lemme know if there's any other info I can give here!

**Comment by snarfed** [comment]
Yup, your sleuthing here matches the description in https://github.com/snarfed/bridgy/issues/217#issuecomment-48375596 , notably that your own replies are the (interment) trigger. Ugh.

I'll keep looking at this. In the meantime, I think you can usually fix it by entering the silo post URL into the "Discover" box on your user page. I tried that for both examples here and it seemed to work.

**Comment by davidcelis** [comment]
Thanks for the workaround! I've tried it on each post that gets broken up and it works every time. I was posting a lot more in the last couple weeks than usual since I was on a trip in Japan, and I was getting more interactions; definitely noticed this issue more often than not at this point, so either it's becoming less intermittent or I'm particularly unlucky 😂
